tests/keys/test_keyinputs.py

```diff
@@ -0,0 +1,105 @@
+import pytest
+import threading
+import socket
+import os
+from pathlib import Path
+
+import zendriver as zd
+import http.server
+import socketserver
+from zendriver import SpecialKeys, KeyModifiers, KeyEvents
+
+
+def find_free_port():
+    """Find a free port to avoid conflicts."""
+    with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
+        s.bind(("", 0))
+        s.listen(1)
+        port = s.getsockname()[1]
+    return port
+
+
+@pytest.fixture(scope="module")
+def http_server():
+    """Start HTTP server once for all tests in this module."""
+    PORT = find_free_port()
+    Handler = http.server.SimpleHTTPRequestHandler
+
+    # Change to the directory containing your HTML files
+    original_dir = os.getcwd()
+    test_dir = Path(__file__).parent
+    os.chdir(test_dir)  # Assuming your HTML files are in the same directory as the test
+
+    try:
+        # Create and start server
+        httpd = socketserver.TCPServer(("", PORT), Handler)
+        server_thread = threading.Thread(target=lambda: httpd.serve_forever())
+        server_thread.daemon = True
+        server_thread.start()
+
+        print(f"Server started at http://localhost:{PORT}")
+
+        # Wait briefly to ensure server is running
+        import time
+
+        time.sleep(1)
+
+        yield {"httpd": httpd, "port": PORT, "thread": server_thread}
+
+    finally:
+        # Cleanup
+        print("Shutting down HTTP server...")
+        httpd.shutdown()
+        httpd.server_close()
+        server_thread.join(timeout=5)
+        os.chdir(original_dir)
+        print("Server shut down.")
```

Comment:
There is no need for an http server, just use `sample_file` function like it's done at other places.

Reply:
I made the change but I cannot push them to your fork directly. Can you cherry pick `f77cdb2` (from the original repo branch [hvgupta/main](https://github.com/stephanlensky/zendriver/compare/hvgupta/main))?